### PR TITLE
feat: Added profile image delete button

### PR DESCRIPTION
**Pull request by sebiweise**
## What does this PR do?

Add a button to delete your current profile picture
image

## How should this be tested?

- Add a profile image
- Check if the delete button is visible
- Delete your profile image

## Checklist

- [x] Add a button to delete your current profile image
- [x] Add a action that will handleRemove of the profile image
- [ ] Add a function that will delete the S3 object

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/web/app/(app)/environments/[environmentId]/settings/profile/actions.ts [hunk 1]
```diff
@@ -82,3 +82,17 @@ export async function updateAvatarAction(avatarUrl: string) {
...
+  const session = await getServerSession(authOptions);
+
+  if (!session) {
+    throw new Error("Not authenticated");
+  }
+
+  if (!session.user.id) {
+    throw new Error("User not found");
+  }
+
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/settings/profile/actions.ts:96]: Can we here also delete the image from the storage?
For this we have a `deleteFile` method in `packages/lib/storage/service.ts`
But then we also need to check if the current user has environment access:
```
const isUserAuthorized = await hasUserEnvironmentAccess(session.user.id, environmentId);
```

**Review by mattinannt — COMMENTED** [review]
@sebiweise thanks for the PR 😊🚀💪
Makes sense and looks good :-) Only one suggestion so we don't clutter our storage too much:

**Comment by jobenjada** [comment]
Hey @sebiweise - mind shipping the little tweak so we can get this merged? 🤓

File: apps/web/app/(app)/environments/[environmentId]/settings/profile/actions.ts (hunk 1 above)
Comment (sebiweise) [reply apps/web/app/(app)/environments/[environmentId]/settings/profile/actions.ts:96]:
```
export async function removeAvatarAction() {
  const session = await getServerSession(authOptions);
  if (!session) throw new AuthorizationError("Not authorized");

  if (!session.user.id) {
    throw new Error("User not found");
  }

  const isUserAuthorized = await hasUserEnvironmentAccess(session.user.id, environmentId);
  if (!isUserAuthorized) throw new AuthorizationError("Not authorized");

  const { message, success, code } = await deleteFile(environmentId, accessType, fileName);

  if (success) {
    // revalidate cache
    storageCache.revalidate({ fileKey: `${environmentId}/${accessType}/${fileName}` });
    return responses.successResponse(message);
  }

  if (code === 404) {
    return responses.notFoundResponse("File", "File not found");
  }

  return await updateUser(session.user.id, { imageUrl: null });
}
```

So I think it should look something like this, but do you know how to get the filename? Is it just the last part of the url?

**Comment by sebiweise** [comment]
@Dhruwang Thanks for your changes, but I´m getting a error why deleting my profile picture:
```
Error: Deletion failed
    at b (/var/task/apps/web/.next/server/app/(app)/environments/[environmentId]/settings/profile/page.js:1:38926)
````

**Comment by Dhruwang** [comment]
@sebiweise ,can you check if have the image available in `web/app/uploads/[environmentId]/public` ?😊

**Comment by sebiweise** [comment]
@Dhruwang Yes the profile picture is available in my bucket. Maybe something related to the custom S3 environment?
I´m using StorJ currently.

**Comment by Dhruwang** [comment]
Hey @sebiweise , sorry for the delay. We actually tested it on S3 as well as using StorJ and deletion works fine there
Can you check if you have the correct Access Permissions set for you access key 😊
Screenshot 2024-03-21 at 6 40 25 PM

**Comment by Dhruwang** [comment]
Hello @sebiweise, thanks a lot for your PR! 😊🚀🙌🏻 It was nearly ready to merge, but there were some issues with merge conflicts, and I encountered a 403 error when trying to push to this branch. 
Screenshot 2024-04-08 at 3 35 14 PM


To resolve this, I've opened a new PR here at https://github.com/formbricks/formbricks/pull/2411, which includes your contributions.

**Pull request closed without merging**